mobile/src/full/java/org/openhab/habdroid/core/GcmRegistrationService.java

```diff
@@ -105,13 +107,15 @@ private void registerGcm(CloudConnection connection) throws IOException {
         InstanceID instanceId = InstanceID.getInstance(this);
         String token = instanceId.getToken(connection.getMessagingSenderId(),
                 GoogleCloudMessaging.INSTANCE_ID_SCOPE, null);
-        String deviceModel = URLEncoder.encode(Build.MODEL, "UTF-8");
-        String deviceId =
-                Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID);
+        String deviceName = getDeviceName()
+                + (BuildConfig.FLAVOR.toLowerCase().contains("beta") ? " (Beta)" : "");
+        deviceName = URLEncoder.encode(deviceName, "UTF-8");
+        String deviceId = Settings.Secure.getString(getContentResolver(),
+                Settings.Secure.ANDROID_ID) + BuildConfig.FLAVOR;
```

Comment:
Doing so will invalidate existing registrations, is this intended? IMHO prepending 'beta:' or similar in the beta case should be sufficient.